src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -89,35 +80,82 @@ def validate_address_pool_name_or_id(namespace):
             child_type='backendAddressPools',
             child_name=pool_name)
 
-def validate_subnet_name_or_id(namespace):
-    """ Validates a subnet ID or, if a name is provided, formats it as an ID. """
-    if namespace.virtual_network_name is None and namespace.subnet is None:
-        return
-    if namespace.subnet == '':
-        return
+def validate_address_prefixes(namespace):
 
-    # error if vnet-name is provided without subnet
-    if namespace.virtual_network_name and not namespace.subnet:
-        raise CLIError('You must specify --subnet name when using --vnet-name.')
+    subnet_prefix_set = SPECIFIED_SENTINEL in namespace.subnet_address_prefix
+    vnet_prefix_set = SPECIFIED_SENTINEL in namespace.vnet_address_prefix
+    namespace.subnet_address_prefix = \
+        namespace.subnet_address_prefix.replace(SPECIFIED_SENTINEL, '')
+    namespace.vnet_address_prefix = namespace.vnet_address_prefix.replace(SPECIFIED_SENTINEL, '')
 
-    # determine if subnet is name or ID
-    is_id = is_valid_resource_id(namespace.subnet)
+    if namespace.subnet_type != 'new' and (subnet_prefix_set or vnet_prefix_set):
+        raise CLIError('Existing subnet ({}) found. Cannot specify address prefixes when '
+                       'reusing an existing subnet.'.format(namespace.subnet))
 
-    # error if vnet-name is provided along with a subnet ID
-    if is_id and namespace.virtual_network_name:
-        raise argparse.ArgumentError(None, 'Please omit --vnet-name when specifying a subnet ID')
-    elif not is_id and not namespace.virtual_network_name:
-        raise argparse.ArgumentError(None,
-                                     'Please specify --vnet-name when specifying a subnet name')
-    if not is_id:
-        namespace.subnet = resource_id(
-            subscription=get_subscription_id(),
+def validate_cert(namespace):
+
+    params = [namespace.cert_data, namespace.cert_password]
+    if all([not x for x in params]):
+        # no cert supplied -- use HTTP
+        namespace.http_listener_protocol = 'http'
+        if not namespace.frontend_port:
+            namespace.frontend_port = 80
+    else:
+        # cert supplied -- use HTTPS
+        if not all(params):
+            raise argparse.ArgumentError(
+                None, 'To use SSL certificate, you must specify both the filename and password')
+
+        # extract the certificate data from the provided file
+        with open(namespace.cert_data, 'rb') as f:
+            contents = f.read()
+            base64_data = base64.b64encode(contents)
+            try:
+                namespace.cert_data = base64_data.decode('utf-8')
+            except UnicodeDecodeError:
+                namespace.cert_data = str(base64_data)
+
+        try:
+            # change default to frontend port 443 for https
+            if not namespace.frontend_port:
+                namespace.frontend_port = 443
+            namespace.http_listener_protocol = 'https'
```

Comment:
If the user specifies this, it should treat it like an unrecognized argument. I've updated it to do so.

---
In reply to: [74095412](https://github.com/Azure/azure-cli/pull/616#discussion_r74095412) [](ancestors = 74095412)